File: app/src/main/java/com/example/kanban/TaskDao.java
package com.example.kanban;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

/**
 * The database Access Object interface
 * This is used to access the database and is used within the TaskRepository class
 * Is not used directly
 */
@Dao
public interface TaskDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE) void insert(Task task);
    @Update(onConflict = OnConflictStrategy.REPLACE) void update(Task task);
    @Delete void deleteTask(Task task);
    @Query("DELETE FROM project_tasks WHERE `key` = :key;") void deleteTaskQuery(int key);
    @Query("SELECT * FROM project_tasks ORDER BY `key` DESC;") LiveData<List<Task>> getAllTasks();
    @Query("SELECT * FROM project_tasks WHERE `key` = :key;") Task getTask(int key);
    @Query("DELETE FROM project_tasks;") void clearTable();
}

File: app/src/main/java/org/eu/net/zendai/kanbankt/TaskDao.kt
package org.eu.net.zendai.kanbankt

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface TaskDao {
    @Query("SELECT * FROM tasks ORDER BY time DESC") fun getAll(): Flow<List<Task>>
    @Query("DELETE FROM tasks") suspend fun deleteAll()
    @Delete suspend fun delete(task: Task)
    @Insert(onConflict = OnConflictStrategy.REPLACE) suspend fun insertAll(vararg task: Task)
    @Insert(onConflict = OnConflictStrategy.REPLACE) suspend fun insert(task: Task)
    @Update(onConflict = OnConflictStrategy.REPLACE) suspend fun update(task: Task)
}